pkg/scheduler/objects/application.go

```diff
@@ -827,13 +827,41 @@ func (sa *Application) tryPlaceholderAllocate(nodeIterator func() NodeIterator,
 			if ph.released || request.taskGroupName != ph.taskGroupName {
 				continue
 			}
+			// before we check anything we need to check the resources equality
+			delta := resources.Sub(ph.AllocatedResource, request.AllocatedResource)
+			// Any negative value in the delta means that at least one of the requested resource in the real
+			// allocation is larger than the placeholder. We need to cancel the placeholder and check the next
+			// placeholder. This should trigger a cleanup of all the placeholder as a task group is always one size.
```

Comment:
> This should trigger a cleanup of all the placeholder as a task group is always one size.

It sounds to me that all other placeholders will be removed (= cleaned up)?

Reply:
I would rephrase it as "task group has a size of one", sounds more correct.

Reply:
Updated comment to be clearer